### PR TITLE
[WFLY-5332]: Allow configure max-threads and core-threads independently. Allow idle thread reusage.

### PR DESCRIPTION
**Pull request by panossot**
[WFLY-5332] : https://issues.jboss.org/browse/WFLY-5332

The PR will fail till https://github.com/jbossas/jboss-threads/pull/19 is included.

### CODE REVIEW

File: threads/src/main/java/org/jboss/as/threads/ThreadsParser.java [hunk 1]
```diff
@@ -512,6 +513,12 @@ public String parseUnboundedQueueThreadPool(final XMLExtendedStreamReader reader
                     foundMaxThreads = true;
                     break;
                 }
+                case CORE_THREADS: {
```
Comment (ctomc) [inline threads/src/main/java/org/jboss/as/threads/ThreadsParser.java]: introducing new attribute to xsd schema requires new version of xsd

Comment (ctomc) [inline threads/src/main/java/org/jboss/as/threads/ThreadsParser.java]: never mind, it looks like it is bug in old version of parser.
2.0 parser handles is correctly as is.

Comment (panossot) [inline threads/src/main/java/org/jboss/as/threads/ThreadsParser.java]: I will check it.

Comment (panossot) [inline threads/src/main/java/org/jboss/as/threads/ThreadsParser.java]: New version of xsd is added.

**Comment by bstansberry** [comment]
Closing to clean up the queue, as this won't compile. Please reopen if the library change this depends on is available.

**Pull request closed without merging**